frontend/src/components/cards/authCard/AuthCard.tsx

```diff
@@ -236,7 +240,20 @@ export default function AuthCard(props: iAuthCard) {
         )
     }
 
+    function renderLoading() {
+        return (
+            <div className="auth-card" style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center' }}>
+                <div className="spinner">
+                    <Spinner animation="border" color="#f8f4e3" />
+                </div>
+            </div>
+        )
+    }
+
     return (
-        props.type === "login" ? renderLogin() : props.type === "signup" ? renderSignUp() : <div> Insert correct prop type</div>
+        APIStatusCode !== 200 ?
+            props.type === "login" ? renderLogin() : props.type === "signup" ? renderSignUp() : <div> Insert correct prop type</div>
+            :
+            renderLoading()
```

Comment:
Render loading spinner as the UI transitions from successfully authenticating to showing the user's profile page. We don't want to show a blank login / signup while UI loads.